### ISSUE TITLE
How do we evaluate trained models on the test set?

### ISSUE THREAD

**Issue by HLasse** [Aarhus-Psychiatry-Research/psycop-common#523]
**Desired behaviour:**
* We want to be able to evaluate trained models on the test set/external data sets and access information from the trained pipe such as feature importance. 
* This requires saving (at a minimum)
    * The fitted model pipeline
    * The preprocessing pipeline

**Possible solutions**
Two things need to happen
1. Saving the necessary information
2. Writing evaluation code

1) could be solved by saving the whole `Trainer` to disk, as it stores both the fitted model pipeline and preprocessing steps. Alternatively, we could save the `Task` and `PreprocessingPipeline` separately and patch them back together

2) could be solved by writing a `.evaluate` method to each Trainer. Is probably ideal, since the CV trainer should refit the model on the full training data before applying it to the test set, whereas this behaviour is not needed for the split trainer (though `refit_on_full_data` could just be a parameter). Alternatively, we could write an evaluation function that takes a `Task`, `PreprocessingPipeline` and the test data and simply applies the trained model.  

**Proposed solution**
Adding the option to save a dataclass containing the PreprocessingPipeline and the Task immediately after training. Writing an evaluation function that applies the preprocessing steps and the model pipe to new data. 

@KennethEnevoldsen @MartinBernstorff thoughts?

**Comment by HLasse** [comment]
> 0. does model pipeline include e.g. z-score normalization? Assuming it does we can just save the pipeline. That seems easy enough. A folder with a pickle dump of the model + a config, potentially some performance metrics from training.

It can, but doesn't have to. The `Task` includes a lot of logic for inference so it would be nifty to save too.

> 2.2) I don't think it should retrain the model by default, but it should be an option to do it in a workflow.

Agreed
